src/lib/prov/bearssl/bearssl_ec.cpp

```diff
@@ -0,0 +1,194 @@
+/*
+* ECDSA via BearSSL
+* (C) 2015,2016 Jack Lloyd
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/hash.h>
+#include <botan/scan_name.h>
+#include <botan/internal/bearssl.h>
+
+#if defined(BOTAN_HAS_ECC_PUBLIC_KEY_CRYPTO)
+  #include <botan/der_enc.h>
+  #include <botan/pkcs8.h>
+  #include <botan/oids.h>
+  #include <botan/internal/pk_ops_impl.h>
+#endif
+
+#if defined(BOTAN_HAS_ECDSA)
+  #include <botan/ecdsa.h>
+#endif
+
+extern "C" {
+  #include <bearssl_hash.h>
+  #include <bearssl_ec.h>
+}
+
+namespace Botan {
+
+#if defined(BOTAN_HAS_ECC_PUBLIC_KEY_CRYPTO)
+
+namespace {
+
+int BearSSL_EC_curve_for(const EC_Group& group)
+   {
+   if(group == EC_Group("secp256r1"))
+      return BR_EC_secp256r1;
+   if(group == EC_Group("secp384r1"))
+      return BR_EC_secp384r1;
+   if(group == EC_Group("secp521r1"))
+      return BR_EC_secp521r1;
+
+   return -1;
+   }
+
+const br_hash_class *BearSSL_hash_class_for(const std::string& emsa)
+   {
+   if (emsa == "EMSA1(SHA-1)")
+      return &br_sha1_vtable;
+   if (emsa == "EMSA1(SHA-224)")
+      return &br_sha224_vtable;
+   if (emsa == "EMSA1(SHA-256)")
+      return &br_sha256_vtable;
+   if (emsa == "EMSA1(SHA-384)")
+      return &br_sha384_vtable;
+   if (emsa == "EMSA1(SHA-512)")
+      return &br_sha512_vtable;
+
+   return NULL;
+   }
+}
+
+#endif
+
+#if defined(BOTAN_HAS_ECDSA) && !defined(OPENSSL_NO_ECDSA)
```

Comment:
Correct, it's one library with all the stuff inside, so there's no disable button. Will correct it.